vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewSAXImporter.java

```diff
@@ -1247,6 +1248,21 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
         }
     }
 
+    private int getSiblWithReqPrimType(Node child) throws RepositoryException {
```

Comment:
This IMHO not only returns the number of siblings but includes child itself. Also the method does not care about required primary type. Why not rename to `getNumChildrenOfPrimaryNodeType(Node parent, String primaryNodeType)`

Reply:
or:

```
getNumChildrenWithType(Node parent, String primaryNodeType)
```

and then check using `isNodeType()` against the required type (not the child type), since the child you are testing could be an super type of the one that is required by the CND.